### DISCUSSION TITLE
out of memory problem

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#749] [Help]
I just got an "out of memory" error when loading my main mindmap. Thankfully, the content is still there. I can load it just fine in a text editor, so I can painstakingly create a new mindmap and extract the content from the text display of the .mm file. But I'd really rather not have to do that, obviously.

Anything you can suggest so I don't have to go through the massive extraction process would be most appreciated.

Big lesson I need to learn again and again and again - create backup copies often. Every day minimum, and whenever I've done a significant chunk of work. 

Here's the message I got. I pressed Send, so the Freeplane team should have that report.

![out-of-memory-errormessage](https://user-images.githubusercontent.com/8197883/195955874-031c3a35-867a-4c69-b5d8-285a581173d0.png)

~ Bal

**Comment by wordmuse** [comment]
Is there any utility program that can open a .mm file and strip it of everything but content and links?

**Reply by macmarrum** [reply]
https://docs.freeplane.org/attic/old-mediawiki-content/Troubleshooting.html#freeplane-throws-out-of-memory-exception

 In essence, change the standard `-Xmx512m` to e.g. `-Xmx1g`. If you are launching your application via `freeplane.exe`, you'll find it in `freeplane.l4j.ini` in your installation folder, next to `freeplane.exe`.

**Comment by wordmuse** [comment]
Thank you so very much, macmarrum! You're a lifesaver!

Best regards,
~ Bal

**Comment by wordmuse** [comment]
Because I was able to load my file again, I started porting it, in chunks, to a pristine new mindmap. When I finished, the new mindmap, which has the exact same content, is far, far smaller in size than the original. The new mindmap contains no clones - only copies where they are clones in the original. Otherwise, I think the content is identical.

![why-the-different-sizes](https://user-images.githubusercontent.com/8197883/196011749-a571554c-d86a-4880-8345-886382fe68ab.png)

The question I have is what might cause the new mindmap to be like 10x smaller than the original? Do FP mindmaps tend to "bloat" as nodes are created, changed and deleted? If so, is there a way to clear out the debris of a enlarged mindmap so that I don't have to lose the clones by copying and pasting to a new mindmap?

**Reply by dpolivaev** [reply]
It looks like there is a bug in Freeplane which I can not find without your help.
And for this, I need both mind map files. I think that obfuscated and zipped versions could be enough, I can look at them first.

But if you do not send me the files for analysis it makes me quite sad, because the only interpretation is that you do not consider me to be a trustworthy person. Why?

If you decide to send me the files you can create a private issue at https://gitlab.com/freeplane/freeplane/confidential-issues where you can register using your GitHub account.

**Reply by dpolivaev** [reply]
Don't forget to zip the files, I think that it significantly reduces their size.

**Comment by wordmuse** [comment]
Hi Dimitry,

I can send you an obfuscated file. The actual file is a personal journal,
spanning 5 years. It contains a great deal of personal information, created
under the assumption that it would be private.

It is almost 3:30am where I am now - long past my bedtime. I will send the
obfuscated files, based on the original and the new files, when I awaken.

And - I will zip the files.

My reluctance to send it has nothing to do with whether I consider you a
trustworthy person or not. I *do* consider you trustworthy. But I wouldn't
send this file to my brother or anyone else. The only one who is privy to
it is my wife. I hope you can understand, but if not, I'll have to accept
that.

Regards,
~ Bal

**Reply by dpolivaev** [reply]
Thank you for your kind response, I am looking forward to getting your obfuscated files.

Regards, Dimitry

**Comment by wordmuse** [comment]
Hi Dimitry,

Here you go. Let me know if you have trouble downloading it.
https://1drv.ms/u/s!Ap1Re2GTtkM5h78LuNaJ8QFIe4ZTbw?e=GXqUG6

An observation you might find interesting.

The size of the obfuscated original increased in size with the obfuscation
effort.  The size of the newer file, with the exact same content, got
smaller with the obfuscation. There is something about the original file
that made it get bigger and bigger, as you can see from this screen capture.

[image: image.png]

I truly hope this helps your efforts.

Regards,
~ Bal

**Reply by dpolivaev** [reply]
Yes, it helps a lot. You got the same problem as described in https://github.com/freeplane/freeplane/issues/601
To repair your original file you can open it in a text editor and remove all lines between
`<hook NAME="presentations">` in line 555 and the matching `</hook>` in line 1933868
as well as `<hook NAME="presentations">` in line 1948134 and the matching `</hook>` in line 3881447

Actually, I do not understand why we got this element twice.

Given your example, I can hopefully fix it.

**Comment by wordmuse** [comment]
I'm glad I could be of use to you. I'm mainly a taker in the community,
since I'm not a programmer.
It's good to be able to give a hand.

Best to you, Dimitry.
~ Bal

**Comment by wordmuse** [comment]
Thank you, also, for telling me how to repair the original file. I've now
done so, and have access to my cloned nodes again.
Much appreciated, Dimitry.

Regards,
~ Bal
